### PR TITLE
[Merged by Bors] - feat: 2652 add p package option to smdk to support cargo workspace projects

### PR DESCRIPTION
**Pull request by superstator**

### CODE REVIEW

File: crates/smartmodule-development-kit/src/package.rs [hunk 1]
```diff
@@ -0,0 +1,109 @@
+use std::fmt::Debug;
+use clap::Parser;
+use std::{env};
+use std::path::{Path, PathBuf};
+use anyhow::Result;
+use cargo_metadata::{CargoOpt, MetadataCommand};
+use convert_case::{Case, Casing};
+use std::ffi::OsString;
+use std::fs::read_dir;
```
Comment (sehz) [inline crates/smartmodule-development-kit/src/package.rs:9]: Just minor nit.  Prefer these conventions for import
```
<std>

<third_party>

<fluvio crates>
<local crate>
<and everything else>
```

File: crates/smartmodule-development-kit/src/package.rs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+pub struct PackageInfo {
+    // The requested package/project name
+    pub package: String,
+    // Inferred binary output path
+    pub output_path: PathBuf,
+}
+impl PackageInfo {
+    /// Finds the closes Cargo.toml in the tree, starting from the current directory
+    pub fn get_current_project_path() -> Option<PathBuf> {
+        let cwd = env::current_dir().expect("Failed to get current working directory");
```
Comment (sehz) [inline crates/smartmodule-development-kit/src/package.rs:32]: let's remove expect

File: crates/smartmodule-development-kit/src/package.rs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        let parents = cwd.as_path().ancestors();
+
+        for path in parents {
+            if let Some(filename) = read_dir(path)
+                .expect("Failed to read directory")
```
Comment (sehz) [inline crates/smartmodule-development-kit/src/package.rs:37]: remove expect

File: crates/smartmodule-development-kit/src/package.rs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                .map(|p| p.unwrap().file_name())
+                .find(|p| p.eq(&OsString::from("Cargo.toml")))
+            {
+                return Some(Path::new(path).join(filename));
+            }
+        }
+        None
+    }
+
+    pub fn from_options(options: &PackageOption) -> Result<PackageInfo, String> {
```
Comment (sehz) [inline crates/smartmodule-development-kit/src/package.rs:48]: add more description to what this fn do

**Review by sehz — CHANGES_REQUESTED** [review]
Seem overall look ok.  Few minor nits

**Review by EstebanBorai — APPROVED** [review]
Other than @sehz comments, LGTM!

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**